packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -882,4 +893,20 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
     EnableDeletionProtection: false,
     ...options,
   });
+
+  /**
+   * Get the resource generator based on the type name definition of model directive
+   * The project level strategy will be returned if type name is not found in the map
+   * @param typeName type name definition of model directive
+   * @returns datasource provision type from map.
+   */
+  private getResourceGenerator = (typeName: string): ModelResourceGenerator | undefined => {
+    const datasourceType = this.modelToDatasourceProvisionTypeMap.get(typeName) ?? DDB_DATASOURCE_TYPE;
```

Comment:
Does the `modelToDatasourceProvisionTypeMap` include SQL models with correct `dbType` i.e MySQL and Postgres? If not this will treat all models as DDB.

Reply:
Yes, the map is created during ingestion.